### ISSUE TITLE
Support for the Manifest Deployer

### ISSUE THREAD

**Issue by vangourd** [johbo/k0s-nix#8]
I'd like to see an option to add K8S manifests from Nix config to the k0s cluster using the Manifest Deployer function.

https://docs.k0sproject.io/stable/manifests/

I'm hoping to learn some more and try to add support for this.

**Comment by vangourd** [comment]
https://github.com/johbo/k0s-nix/pull/12

**Comment by johbo** [comment]
Seems that `k0s` is similar in this regard, as commented on the PR at https://github.com/johbo/k0s-nix/pull/12#issuecomment-2495554923: There is this concept of a "Manifest Deployer".

The docs have a nice example, we could use this as a test case: https://github.com/johbo/k0s-nix/pull/12#issuecomment-2495554923

Helm charts seem to work by being configured within the k0s configuration, this is already possible: https://github.com/johbo/k0s-nix/blob/f711400587df6ca465a15f357b04eceacd718a97/nixos/extensions.nix#L33

**Comment by johbo** [comment]
Did look into this topic again, by now I am quite confident that we should have the multi stack option which is in the comment on PR 12: https://github.com/johbo/k0s-nix/pull/12#issuecomment-2495554923

Also looked into the related feature request in #24 about ordering via k0s-nix, did close it because this is nothing that k0s-nix should deal with.

From what I understand about the manifest deployer:

- A "stack" is a directory, and the yaml files inside
- k0s seems to reconcile when the content changes, unclear to me what this does when a nixos-rebuild switch also restarts the k0s unit at the same time. This may lead to some odd situations.
- k0s-nix should IMO only be concerned with getting the files / content there. The rest is on k0s.

**Comment by johbo** [comment]
After digging deeper into it, I think this topic easily is a rabbit hole. My take is to close this as not planned.

What I found is that k0s does apply the manifests out of the folder. And it does drop its own manifests in there, so there is also a naming clash risk. A prefix could help to work around this.

The next aspect is the cleanup: It seems that it watches the stack folders, on removal it seems to delete them if k0s is running while it is removed. If not, then on startup it does not seem to remove them. 

Now I thought that we could provide something useful by just supporting to add the files, but that's also not good: Removal would be manual, forgetting the declarative adjustment would then restore the manifests easily by the next nixos-rebuild switch or even after a reboot. 

With the above I realized that this is a bit more involved, and that using this feature can lead to surprises. On my end I concluded to stay using features like plain kubectl or other means to push manifests into the cluster instead of using the manifest / stack folder of k0s.

There is also the option to call something like `k0s kubectl apply -k <dir>` on the controller which should always work.

**Comment by johbo** [comment]
Here is one of the tests I did to better understand the behavior: https://github.com/johbo/k0s-nix/commit/b7422c143a18c2cc92c8181df432ea56cbfeade2

Might be helpful in case anyone wants to look into this topic.

**Comment by vangourd** [comment]
Yeah I think your analysis is pragmatic. One can use Nix to configure a systemd unit to provision the resources and then the results are easily inspected with journalctl.
